EventSubscriber/UserProfileSubscriber.php

```diff
@@ -43,35 +43,88 @@ public function loadUserPreferences(UserPreferenceEvent $event)
                 ->setOptions(['label'    => 'Target Weekly Start Date' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
         );
 
         $event->addPreference(
-            (new UserPreference('yearly-fte-vacation-days', 35))
-                ->setValue(35)
+            (new UserPreference('yearly-vacation-hours', 168))
+                ->setEnabled($isAdmin)
                 ->setType(NumberType::class)
-                ->setOptions(['label' => 'Yearly Vacation Days' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
+                ->setOptions(['label' => 'Yearly Vacation Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
         );
 
         $event->addPreference(
             (new UserPreference('start-of-period-vacation-hours', 0))
-                ->setValue(0)
+                ->setEnabled($isAdmin)
                 ->setType(NumberType::class)
                 ->setOptions(['label' => 'Start of Period Vacation Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
-            );
+        );
+
+        $event->addPreference(
+            (new UserPreference('extra-vacation-hours', 0))
+                ->setEnabled($isAdmin)
+                ->setType(IntegerType::class)
+                ->setOptions(['label' => 'Extra Vacation Hours' . ($isAdmin ? '' : ' (Read-Only)'), 'attr' => ['readonly' => !$isAdmin]]) // Make readonly if not admin
+        );
     }
 
     public function loadUserProfile(PrepareUserEvent $event)
     {
+        $isAdmin = $this->security->isGranted('ROLE_ADMIN');
+
+        if(!$isAdmin) {
+          // if user is not admin, we show them an unsubmittable fake input, so they know their vacation information
+          echo "<script>
+          async function showVacationInformation() {
+            // for some reason this script is included twice in the page, so we break if its already loaded
+            if (window.DS_VacationData) return;
+
+            window.DS_VacationData = {
+                'Target Weekly Hours': " . $event->getUser()->getPreferenceValue('target-weekly-hours') . ",
+                'Target Weekly Start': " . $event->getUser()->getPreferenceValue('target-weekly-start') . ",
+                'Yearly Vacation Hours': " . $event->getUser()->getPreferenceValue('yearly-vacation-hours') . ",
+                'Start of Period Vacation Hours': " . $event->getUser()->getPreferenceValue('start-of-period-vacation-hours') . ",
+                'Extra Vacation Hours': " . $event->getUser()->getPreferenceValue('extra-vacation-hours') . ",
+            };
+
+            let form;
+            while(!form) {
+              await new Promise(resolve => setTimeout(resolve, 10));
+              form = document.querySelector('form');
+            }
```

Comment:
This could run *forever*, and we don't want that. Instead, run `showVacationInformation` when [the document finishes loading](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event)

Reply:
Done: https://github.com/DelftSolutions/kimai-vacation-hours/pull/4/commits/363b8c31616b5424fb431b1ff78b312d0f669d5e

Turned out window's `load` event also does the job ✅ (which is more familiar for me)

Reply:
They're not the same.

See: https://javascript.info/onload-ondomcontentloaded

You only need the DOM to exist